src/main/java/com/bookstore/service/UserDetailsServiceImpl.java

```diff
@@ -0,0 +1,22 @@
+package com.bookstore.service;
+
+import com.bookstore.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.userdetails.UserDetails;
+import org.springframework.security.core.userdetails.UserDetailsService;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+
+@Service
+@RequiredArgsConstructor
+public class UserDetailsServiceImpl implements UserDetailsService {
+    private final UserRepository userRepository;
+
+    @Override
+    public UserDetails loadUserByUsername(String username)
+            throws UsernameNotFoundException {
+        return userRepository.findByEmailAndDeletedFalse(username)
```

Comment:
```suggestion
        return userRepository.findByEmail(username)
```
implement soft-deleted in the right way